Tests/DependencyInjection/Compiler/AddProcessorsPassTest.php

```diff
@@ -31,12 +32,28 @@ public function testHandlerProcessors()
         $service = $container->getDefinition('monolog.handler.test');
         $calls = $service->getMethodCalls();
         $this->assertCount(1, $calls);
-        $this->assertEquals(['pushProcessor', [new Reference('test')]], $calls[0]);
+        $this->assertEquals(['pushProcessor', [new TypedReference('test', 'TestClass')]], $calls[0]);
 
         $service = $container->getDefinition('handler_test');
         $calls = $service->getMethodCalls();
         $this->assertCount(1, $calls);
-        $this->assertEquals(['pushProcessor', [new Reference('test2')]], $calls[0]);
+        $this->assertEquals(['pushProcessor', [new TypedReference('test2', 'TestClass')]], $calls[0]);
+
+        $service = $container->getDefinition('monolog.handler.priority_test');
```

Comment:
Can we add a case when the processor has a channel?

Reply:
Done